jobs/purge_completed_sessions.go

```diff
@@ -10,8 +11,12 @@ import (
 	"gopkg.in/segmentio/analytics-go.v3"
 )
 
+var purgeMutex = &sync.Mutex{}
+
 func PurgeCompletedSessions(PrometheusWrapper services.PrometheusService) {
 
+	purgeMutex.Lock()
+
 	start := PrometheusWrapper.TimeNow()
 	defer PrometheusWrapper.HistogramSeconds(PrometheusWrapper.HistogramPurgeCompletedSessions, start)
 
```

Comment:
Hmm. But the Prometheus stuff reports on how long something takes, right? It might not be good to start the timer for a new thread, and then make it wait before it can start using the resources. It would make it seem like it takes longer than it does.  

I will add that defer though.

Reply:
True, but it might be useful to know if the threads are getting backed up and include locking as part of the performance metric.

Reply:
Ok I see what you're going for.  I will move it.